src/wallet/wallet.cpp

```diff
@@ -3140,22 +3140,18 @@ bool CWallet::TopUpKeyPool(unsigned int kpSize)
                 internal = true;
             }
 
-            if (!setInternalKeyPool.empty()) {
-                nEnd = *(setInternalKeyPool.rbegin()) + 1;
-            }
-            if (!setExternalKeyPool.empty()) {
-                nEnd = std::max(nEnd, *(setExternalKeyPool.rbegin()) + 1);
-            }
+            int64_t index = ++m_max_keypool_index;
+            assert(index >= 0); // How in the hell did you use so many keys?
```

Comment:
Would be better (non-UB, but also more clear to read) to check for max before the increase.